apps/explorer/src/ui/utils/LinkWithQuery.tsx

```diff
@@ -30,8 +31,12 @@ export function useNavigateWithQuery() {
 
 export const LinkWithQuery = forwardRef<HTMLAnchorElement, LinkProps>(
     ({ to, ...props }) => {
-        const { search } = useLocation();
+        const [searchParams] = useSearchParams();
 
-        return <Link to={`${to}${search}`} {...props} />;
+        const networkParam =
```

Comment:
I’d construct a new url params to build this search param rather than doing it manually.

Reply:
Yes - I've also updated it further such that it should be resilient if further search queries are added to the URL string.